src/MultiFormats/MultiHash.cs

```diff
@@ -0,0 +1,587 @@
+        public void Write(Stream stream)
+        {
+            using (var cos = new CodedOutputStream(stream, true))
+            {
+                Write(cos);
+            }
+        }
+
+        /// <summary>
+        ///   Writes the binary representation of the multihash to the specified <see cref="Google.Protobuf.CodedOutputStream"/>.
+        /// </summary>
+        /// <param name="stream">
+        ///   The <see cref="Google.Protobuf.CodedOutputStream"/> to write to.
+        /// </param>
+        /// <remarks>
+        ///   The binary representation is a <see cref="Varint"/> of the <see cref="HashingAlgorithm.Code"/>,
+        ///   <see cref="Varint"/> of the <see cref="HashingAlgorithm.DigestSize"/> followed by the <see cref="Digest"/>.
+        /// </remarks>
+        public void Write(CodedOutputStream stream)
+        {
+            if (stream == null)
+                throw new ArgumentNullException("stream");
+
+            stream.WriteInt32(Algorithm.Code);
+            stream.WriteLength(Digest.Length);
+            stream.WriteSomeBytes(Digest);
+        }
+
+        private void Read(Stream stream)
+        {
+            using (var cis = new CodedInputStream(stream, true))
+            {
+                Read(cis);
+            }
+        }
+
+        private void Read(CodedInputStream stream)
+        {
+            var code = stream.ReadInt32();
+            var digestSize = stream.ReadLength();
+
+            HashingAlgorithm.Codes.TryGetValue(code, out var a);
+            Algorithm = a;
+            if (Algorithm == null)
+            {
+                Algorithm = HashingAlgorithm.Register("ipfs-" + code, code, digestSize);
+                RaiseUnknownHashingAlgorithm(Algorithm);
+            }
+            else if (Algorithm.DigestSize != 0 && digestSize != Algorithm.DigestSize)
+            {
+                throw new InvalidDataException(string.Format("The digest size {0} is wrong for {1}; it should be {2}.",
+                    digestSize, Algorithm.Name, Algorithm.DigestSize));
+            }
+
+            Digest = stream.ReadSomeBytes(digestSize);
+        }
+
+        /// <inheritdoc />
+        public override int GetHashCode() { return ToString().GetHashCode(); }
+
+        /// <inheritdoc />
+        public override bool Equals(object obj)
+        {
+            var that = obj as MultiHash;
+            return that == null
+                ? false
+                : Equals(that);
+        }
+
+        /// <inheritdoc />
+        public bool Equals(MultiHash that)
+        {
+            return Algorithm.Code == that.Algorithm.Code
+             && Digest.SequenceEqual(that.Digest);
+        }
+
+        /// <summary>
+        ///   Value equality.
+        /// </summary>
+        public static bool operator ==(MultiHash a, MultiHash b)
+        {
+            if (ReferenceEquals(a, b)) return true;
+            if (a is null) return false;
+            if (b is null) return false;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Add curly braces around the nested statement(s) in this 'if' block.](https://app.codacy.com/manual/catalyst-network/Catalyst.Framework/pullRequest?prid=4832783)